From 4b0dbf5a8c4e44d84c0c79237102072d92ecbad2 Mon Sep 17 00:00:00 2001
From: Philippe Renon <philippe_renon@yahoo.fr>
Date: Mon, 12 Feb 2018 15:52:34 +0100
Subject: [PATCH 2/2] cache_rocksdb: use find_package CONFIG mode when building
 with mingw

CONFIG mode makes it simpler to get all needed configuration
compiles fine on mys2/mingw, not sure about other mingw
---
 CMakeLists.txt                                   | 14 ++++++++++++--
 src/osgEarthDrivers/cache_rocksdb/CMakeLists.txt | 18 ++++++++++--------
 2 files changed, 22 insertions(+), 10 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a85db48b3..e8b83736f 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,10 +8,13 @@ if(COMMAND cmake_policy)
     # Works around warnings about escaped quotes in ADD_DEFINITIONS
     # statements.
     cmake_policy(SET CMP0005 OLD)
-    
+
     # Qt5 qt5_use_modules usage was causing "Policy CMP0043 is not set: Ignore COMPILE_DEFINITIONS_<Config> properties." warnings
     cmake_policy(SET CMP0043 NEW)
 
+    # Double colon in target name means IMPORTED (and not ALIAS or IMPORTED).
+    # Fixes a warning when cache_rocksdb links to target "RocksDB::rocksdb"
+    cmake_policy(SET CMP0028 NEW)
 endif(COMMAND cmake_policy)
 
 #
@@ -123,7 +126,14 @@ FIND_PACKAGE(ZLIB)
 FIND_PACKAGE(Poco)
 
 FIND_PACKAGE(LevelDB)
-FIND_PACKAGE(RocksDB)
+IF(MINGW)
+    FIND_PACKAGE(RocksDB CONFIG)
+    IF(RocksDB_FOUND)
+        MESSAGE(STATUS "Found RocksDB " ${RocksDB_VERSION})
+    ENDIF()
+ELSE(MINGW)
+    FIND_PACKAGE(RocksDB)
+ENDIF(MINGW)
 
 FIND_PACKAGE(SilverLining QUIET)
 FIND_PACKAGE(Triton QUIET)
diff --git a/src/osgEarthDrivers/cache_rocksdb/CMakeLists.txt b/src/osgEarthDrivers/cache_rocksdb/CMakeLists.txt
index e3dd9edb0..5b14c6d24 100644
--- a/src/osgEarthDrivers/cache_rocksdb/CMakeLists.txt
+++ b/src/osgEarthDrivers/cache_rocksdb/CMakeLists.txt
@@ -1,6 +1,6 @@
-IF(ROCKSDB_FOUND)
+IF(ROCKSDB_FOUND OR RocksDB_FOUND)
 
-set (CMAKE_CXX_STANDARD 11)
+SET(CMAKE_CXX_STANDARD 11)
 
 INCLUDE_DIRECTORIES( ${ROCKSDB_INCLUDE_DIR} )
 
@@ -8,7 +8,7 @@ SET(TARGET_H
     RocksDBCacheOptions
     RocksDBCache
     RocksDBCacheBin
-	Tracker
+    Tracker
 )
 SET(TARGET_SRC 
     RocksDBCache.cpp
@@ -18,9 +18,13 @@ SET(TARGET_SRC
 
 SET(TARGET_LIBRARIES_VARS ROCKSDB_LIBRARY)
 
-IF(WIN32)
+IF(MSVC)
     SET(TARGET_EXTERNAL_LIBRARIES ws2_32 winmm rpcrt4)
-ENDIF(WIN32)
+ENDIF(MSVC)
+
+IF(MINGW)
+    SET(TARGET_EXTERNAL_LIBRARIES RocksDB::rocksdb)
+ENDIF(MINGW)
 
 SETUP_PLUGIN(osgearth_cache_rocksdb)
 
@@ -29,6 +33,4 @@ SET(LIB_NAME cache_rocksdb)
 SET(LIB_PUBLIC_HEADERS RocksDBCacheOptions)
 INCLUDE(ModuleInstallOsgEarthDriverIncludes OPTIONAL)
 
-ENDIF(ROCKSDB_FOUND)
-
-
+ENDIF(ROCKSDB_FOUND OR RocksDB_FOUND)
-- 
2.16.1

